superdesk/publish/formatters/ninjs_formatter.py

```diff
@@ -311,10 +311,13 @@ def _format_qcodes(self, items):
         return [{'name': item.get('name'), 'code': item.get('qcode')} for item in items]
 
     def _format_place(self, items):
-        locator_map = superdesk.get_resource_service('vocabularies').find_one(req=None, _id='locators')
+        vocabularies_service = superdesk.get_resource_service('vocabularies')
+        locator_map = vocabularies_service.find_one(req=None, _id='locators')
 
         def getLabel(item):
-            locators = [l for l in locator_map.get('items', []) if l['qcode'] == item.get('qcode')]
+            locale_locators = vocabularies_service.get_locale_vocabulary(
+                locator_map.get('items'), item.get('language')) or []
```

Comment:
item here is a value from `item.place` so imo it won't have language set

Reply:
Yes, true
I updated the code